service_crategen/src/commands/generate/codegen/mod.rs

```diff
@@ -160,7 +162,7 @@ where
     // See https://github.com/rusoto/rusoto/issues/519
     writeln!(writer,
              "/// Trait representing the capabilities of the {service_name} API. {service_name} clients implement this trait.
-        pub trait {trait_name} {{
+        pub trait {trait_name}: Clone + Sync + Send + 'static {{
```

Comment:
@srijs these trait bounds was needed to get the default functions returning Streams to work. 

Thesse seem like restrictive additions but in practice I find these usually needed when working with futures. They are already satisfied by the default client impl and the default client impl is typically the only impl for these traits.

Reply:
Looks good to me. An alternative could have been to add the trait bounds on the pagination methods rather than the trait, but I think either is fine.